packages/server/lib/graphql/ServerContext.ts

```diff
@@ -18,6 +19,14 @@ export class ServerContext extends BaseContext {
         ? new Viewer(this, cachedUser)
         : null
     })
+
+    this._getLocalProjectsFromCache()
+  }
+
+  async _getLocalProjectsFromCache () {
```

Comment:
I see one potential problem with this approach; since this is async, it will happen in the background, and we don't have any way for the GraphQL client to know that the cache is still loading. There is no way for urql to know that this has finished.

Rather than populating the project cache in the `ServerContext` constructor, I think we should popular it the first time the `addProject` query is made. This way urql will know when it's asked for the cached projects (via a query) and when they have finished loading, and we can reactively update the UI state accordingly.

If this doesn't make sense (just read my explanation, it's a bit confusing) we can pair on it.

Reply:
I think that makes sense. I think we could make another mutation called loadProjectsFromCache that will populate this field. Then the GlobalPage that needs it can make the call to load the projects with urql.

Reply:
abstracted this logic out to loadProjects mutation